audioplayer.cpp

```diff
@@ -0,0 +1,92 @@
+#include "audioplayer.h"
+
+vector<string> AudioPlayer::GetAvailableTracks(const string& folder) {
+  vector<string> names;
+  DIR* directory = opendir(folder.c_str());
+
+  if (directory == nullptr) {
+	throw invalid_argument("Unable to find directory");
+  }
+
+  dirent* ent;
+  while ((ent = readdir(directory)) != nullptr) {
+	if ((strcmp(ent->d_name, "..")) && (strcmp(ent->d_name, "."))) {
+	  names.emplace_back(ent->d_name);
+	}
+  }
+  closedir(directory);
+  return names;
+}
+
+void AudioPlayer::CloseAudio() {
+  CallMciSendString("close mp3");
+}
+
+bool AudioPlayer::PlayAudio() {
+  if (selectedTrackName != "") {
+	// save current volume
+	CallMciSendString(("setAudio mp3 volume to " + currentVolume));
+
+	// play audio
+	isPlaying = true;
+	CallMciSendString("play mp3");
+
+	return true;
+
+  } else {
+	return false;
+  }
+}
+
+void AudioPlayer::ChangeVolume(const string& volume) {
+  currentVolume = volume;
+  CallMciSendString("setAudio mp3 volume to " + currentVolume);
+}
+
+void AudioPlayer::RewindAudio(const string& trackTiming) {
+  // rewind
+  CallMciSendString(("seek mp3 to " + trackTiming + "000"));
+
+  if (!PlayAudio()) {
```

Comment:
You should choose only one error handling strategy - either exceptions or return true / false. You should fix it everywhere

Reply:
done (have chosen returning true/false)